gh-84436: update docs on Py_None/Py_True/Py_False/Py_Ellipsis becoming immortal (#105195)
python/cpython: Doc/c-api/bool.rst

.. c:var:: PyObject* Py_False

>    The Python ``False`` object.  This object has no methods and is immortal.

Reviewer: Could this link to a glossary entry or docs section on immortal objects? Users may have no idea what this means.

Author: I don't think we have anything in the docs. I can link to the PEP.

Reviewer: Ideally there should be a new section in the docs, but linking to the PEP is fine for now.

.. versionchanged:: 3.12

>    :const:`Py_False` is immortal.

Reviewer: `:c:const:` maybe?

Author: I think I picked up ``:const:`` from the Py_RETURN_FALSE doc. 

Is there such a thing as ``:c:const:`` (grep on the docs dir doesn't give me any).

Reviewer: I think we actually want ":c:data:". That's how the link to Py_None in https://docs.python.org/3.10/c-api/slice.html#c.Py_Ellipsis is created.

Author: I had :c:data: initially but there was some build error. I'll try it again.

.. c:var:: PyObject* Py_False

   The Python ``False`` object.  This object has no methods and is

>    `immortal.<https://peps.python.org/pep-0683/>`_

Author: ```suggestion
   `immortal <https://peps.python.org/pep-0683/>`_.
```

.. versionchanged:: 3.12
   :c:data:`Py_False` is immortal.

.. c:var:: PyObject* Py_True

   The Python ``True`` object.  This object has no methods and is

>    `immortal.<https://peps.python.org/pep-0683/>`_

Author: ```suggestion
   `immortal <https://peps.python.org/pep-0683/>`_.
```

.. versionchanged:: 3.12
   :c:data:`Py_True` is immortal.

.. c:macro:: Py_RETURN_FALSE

>    Return :const:`Py_False` from a function.

Reviewer 2: ```suggestion
   Return :c:data:`Py_False` from a function.
```

.. c:macro:: Py_RETURN_TRUE

>    Return :const:`Py_True` from a function.

Reviewer 2: ```suggestion
   Return :c:data:`Py_True` from a function.
```

.. c:function:: PyObject* PyBool_FromLong(long v)

>    Return :const:`Py_True` or :const:`Py_False`, depending on the truth value of *v*.

Reviewer 2: ```suggestion
   Return :c:data:`Py_True` or :c:data:`Py_False`, depending on the truth value of *v*.
```
